deflate_fast.c

```diff
@@ -95,7 +92,7 @@ Z_INTERNAL block_state deflate_fast(deflate_state *s, int flush) {
         if (UNLIKELY(bflush))
             FLUSH_BLOCK(s, 0);
     }
-    s->insert = s->strstart < MIN_MATCH-1 ? s->strstart : MIN_MATCH-1;
+    s->insert = s->strstart < (WANT_MIN_MATCH - 1) ? s->strstart : (STD_MIN_MATCH - 1);
```

Comment:
I'm not 100% confident this is correct... it is guard for making sure there is at least 1 byte available before the string due to hashing offset.

Reply:
Yes, this is very much a request for comments PR, I am not sure about these either.
Please submit suggestions.